.github/workflows/sphinx.yml

```
@@ -0,0 +1,44 @@
+name: Sphinx Deploy
+
+on:
+  workflow_call:
+  inputs:
+    artifact-name:
+      required: true
+      default: cpp-linter_docs
+      type: string
+      description: The docs artifact name
+    path-to-doc:
+      required: true
+      type: string
+      default: docs/_build/html
+      description: The docs path name
+
+jobs:
+  sphinx-deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+      - name: Install dependencies
+        run: pip install sphinx-immaterial
+
+      - name: Build docs
+        if: github.ref != 'refs/heads/main'
+        run: sphinx-build docs ${{ inputs.path-to-doc }}
+
+      - name: Upload docs build as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ inputs.artifact-name }}
```

No need to use an input for this
```suggestion
          name: ${{ github.repository }}_docs
```